### PR TITLE
Fix BBEs

### PR DESCRIPTION
**Pull request by NatashaWso2**

### CODE REVIEW

File: examples/functions/functions.bal [hunk 1]
```diff
@@ -1,15 +1,19 @@
 import ballerina/io;
 
-// This function takes a `string` argument, but does not have a return value.
+// This function takes in a `string` argument, but does not have a return
+// value.
```
Comment (pubudu91) [inline examples/functions/functions.bal]: How about "does not return a value" instead?

File: examples/functions/functions.bal [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 function printValue(string value) {
     io:println(value);
 }
 
-// This function takes two `int` values and returns their sum as an integer.
+// This function takes in two `integer` values as arguments and returns their
```
Comment (pubudu91) [inline examples/functions/functions.bal]: `int` is fine here right? If using "integer", then let's get rid of the back ticks.

File: examples/functions/functions.bal [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
+// sum as an integer.
 function add(int a, int b) returns (int) {
     return a + b;
 }
 
+// This function named `main` is considered as the default function as an entry
```
Comment (pubudu91) [inline examples/functions/functions.bal]: `main` -> `main()`

File: examples/modules/modules.bal [hunk 4]
```diff
@@ -1,6 +1,9 @@
-// Here's how you can import a module. You can only refer to public symbols of an imported module.
-// By default, the last element of the module name becomes an alias that is used to refer to symbols of the imported module.
-// If the module name has dots, then the last word after the last dot becomes the alias.
+// Here's how you can import a module. You can only refer to public symbols of
+// an imported module.
+// By default, the last element of the module name becomes an alias that is
+// used to refer to symbols of the imported module.
+// If the module name has dots, then the word after the last dot becomes the
```
Comment (pubudu91) [inline examples/modules/modules.bal]: dots -> periods ?

File: examples/functions/functions.bal [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
+// point.
```
Comment (pubudu91) [inline examples/functions/functions.bal]: @MaryamZi what would be the best way to phrase this?

File: examples/modules/modules.description [hunk 6]
```diff
@@ -1,3 +1,3 @@
 // Ballerina program files have the `.bal` extension and the file names do not have semantics.
 // These source files can be grouped into modules. The code within a module can refer to any module-level symbol (name) declared
-// in the same module.
+// in the same module. Module names can only contain alphanumerics, underscores and periods and the maximum length is 256 characters
```
Comment (pubudu91) [inline examples/modules/modules.description]: Missing period at the end

File: examples/modules/modules.bal (hunk 4 above)
Comment (samgnaniah) [reply examples/modules/modules.bal]: +1.

File: examples/variables/variables.bal [hunk 7]
```diff
@@ -6,12 +6,14 @@ int total = 99;
 // Declare a public module-level variable.
 public int count = 0;
 
-// Declare a constant. `const`'s value is frozen. Value is computed at compile time therefore immutable.
+// Declare a constant. Value of the constant is frozen. Value is computed at
+// compile time therefore is immutable.
```
Comment (pubudu91) [inline examples/variables/variables.bal]: Shall we rephrase this? This feels like disconnected sentence fragments.

File: examples/variables/variables.bal [hunk 8]
```diff
@@ -6,12 +6,14 @@ int total = 99;
...
 const string OK = "ok";
 
 // Declare a public final variable that behaves as a constant for other modules.
-// Value of final variable is forzen. All the parameters for a function call is also implicitly final.
-// Value is computed at runtime therefore not immutable but cannot assign any other value since variable is final.
+// Value of the final variable is frozen. All parameters of a function call are
+// also implicitly final. Value is computed at runtime therefore not immutable.
+// Since the variable is final any other value cannot be assigned to it.
```
Comment (pubudu91) [inline examples/variables/variables.bal]: The last two sentences and the second sentence sound contradictory.

Comment (pubudu91) [reply examples/variables/variables.bal]: Shall we get this clarified?

File: examples/functions-with-defaultable-parameters/functions_with_defaultable_parameters.bal [hunk 9]
```diff
@@ -1,13 +1,13 @@
 import ballerina/io;
 
-// This function takes salary, annual increment, and bonus rate as input 
+// This function takes base salary, annual increment and bonus rate as input
```
Comment (samgnaniah) [inline examples/functions-with-defaultable-parameters/functions_with_defaultable_parameters.bal]:
```suggestion
// This function takes the base salary, annual increment, and the bonus rate as input
```

File: examples/functions-with-defaultable-parameters/functions_with_defaultable_parameters.bal [hunk 10]
```diff
@@ -1,13 +1,13 @@
...
 // parameters and prints them. The first parameter `baseSalary` is a required
 // parameter. The next two parameters `annualIncrement` and `bonusRate` are 
-// two defaultable parameters. They are optional parameters with a default 
-// value of 20 and 0.02 respectively. Defaultable parameters must always be
+// two defaultable parameters. They are optional parameters which have default
```
Comment (samgnaniah) [inline examples/functions-with-defaultable-parameters/functions_with_defaultable_parameters.bal]:
```suggestion
// two defaultable parameters. They are optional parameters that have default
```

File: examples/modules/modules.description [hunk 11]
```diff
@@ -1,3 +1,3 @@
 // Ballerina program files have the `.bal` extension and the file names do not have semantics.
 // These source files can be grouped into modules. The code within a module can refer to any module-level symbol (name) declared
-// in the same module.
+// in the same module. Module names can only contain alphanumerics, underscores and periods and the maximum length is 256 characters.
```
Comment (samgnaniah) [inline examples/modules/modules.description]:
```suggestion
// in the same module. Module names can only contain alphanumerics, underscores, and periods and the maximum length is 256 characters.
```

File: examples/var/var.bal [hunk 12]
```diff
@@ -1,7 +1,8 @@
 import ballerina/io;
 
 public function main() {
-    //The variable type is inferred from the initial value. This is the same as `int k = 5`;
+    //The variable type is inferred from the expression on the right-hand side.
+    //This is the same as `int k = 5`;
```
Comment (samgnaniah) [inline examples/var/var.bal]:
```suggestion
    //This is the same as `int k = 5;`.
```

File: examples/variables/variables.description [hunk 13]
```diff
@@ -1,3 +1,3 @@
-// Ballerina allows two ways to declare variables.
+// Ballerina provides you two ways to declare variables.
```
Comment (samgnaniah) [inline examples/variables/variables.description]:
```suggestion
// Ballerina provides you with two ways to declare variables.
```

File: examples/variables/variables.bal (hunk 7 above)
Comment (MaryamZi) [reply examples/variables/variables.bal]: IMO, we don't need to go into details about constants here. 

Ideally the constants BBE should have details about constants always being frozen (immutable), and here we can just say "Declare a constant."

WDYT?

File: examples/variables/variables.bal (hunk 8 above)
Comment (NatashaWso2) [reply examples/variables/variables.bal]: Fixed with d23238e

**Comment by pubudu91** [comment]
Shall we not add fixes to all BBEs in one PR? That makes it harder to review

File: examples/documentation/documentation.bal [hunk 14]
```diff
@@ -5,9 +5,9 @@ public function main() {
 }
 
 // This is the documentation attachment for object `DummyObject`.
-# `DummyObject` is a user defined object type in Ballerina program. This
-# `DummyObject` is used to hold two string data fields and
-# one attached function definition, `doThatOnObject` performs that functionality
+# `DummyObject` is a user defined object type in your Ballerina program.
+# This `DummyObject` is used to hold two string data fields and one attached
```
Comment (samgnaniah) [inline examples/documentation/documentation.bal]:
```suggestion
# This `DummyObject` is used to hold two `string` data fields and one attached
```

File: examples/documentation/documentation.description [hunk 15]
```diff
@@ -1,2 +1,2 @@
-// Documentation attachments allows you to document different Ballerina constructs such as functions, records, objects, endpoints etc.
-// These attachments will be used by Docerina to generate API documentation for your modules.
\ No newline at end of file
+// Documentation attachments allows you to document different Ballerina constructs such as functions, records, objects, endpoints etc. defined in your modules.
```
Comment (samgnaniah) [inline examples/documentation/documentation.description]:
```suggestion
// Documentation attachments allow you to document different Ballerina constructs such as functions, records, objects, endpoints, etc. defined in your modules.
```

File: examples/documentation/documentation.description [hunk 16]
```diff
@@ -1,2 +1,2 @@
...
+// These attachments will be used by Docerina (Doc Generation Tool) to generate API documentation for your modules.
```
Comment (samgnaniah) [inline examples/documentation/documentation.description]:
```suggestion
// These attachments are used by Docerina (Documentation Generation Tool) to generate API documentation for your modules.
```

File: examples/functions-with-defaultable-parameters/functions_with_defaultable_parameters.bal (hunk 9 above)
Comment (samgnaniah) [inline examples/functions-with-defaultable-parameters/functions_with_defaultable_parameters.bal]:
```suggestion
// This function takes base salary, annual increment, and bonus rate as input
```

The serial comma usage is correct here actually. You would normally have the comma in front of "and" in a list.

File: examples/functions-with-defaultable-parameters/functions_with_defaultable_parameters.bal (hunk 10 above)
Comment (samgnaniah) [inline examples/functions-with-defaultable-parameters/functions_with_defaultable_parameters.bal]:
```suggestion
// two defaultable parameters. They are optional parameters that have default
```

You would normally use "that" in this context. The word "which" is used when you are adding additional information to the sentence that is not necessary for the sentence to be complete. On the other hand, "that" is used to add necessary content for sentence completion. Also, when you use "which" in this context, you would normally have a comma in front of it.

File: examples/functions-with-defaultable-parameters/functions_with_defaultable_parameters.bal [hunk 17]
```diff
@@ -16,23 +16,24 @@ function printSalaryDetails(int baseSalary,
...
+    // Call the function by passing only the `baseSalary`.
+    // The `annualIncrement` and the `bonusRate` defaults to 20 and 0.02
+    // respectively.
     printSalaryDetails(2500);
 
-    // Call the function by passing only the `salary` and `annualIncrement`. 
-    // Pass the defaultable parameter as a key-value pair, when
-    // invoking the function. The `bonusRate` defaults to 0.02.
+    // Call the function by passing only the `baseSalary` and `annualIncrement`.
+    // Pass the defaultable parameter as a key-value pair, when invoking the
```
Comment (samgnaniah) [inline examples/functions-with-defaultable-parameters/functions_with_defaultable_parameters.bal]:
```suggestion
    // Pass the defaultable parameter as a key-value pair when invoking the
```

File: examples/functions-with-rest-parameter/functions_with_rest_parameter.bal [hunk 18]
```diff
@@ -1,8 +1,9 @@
 import ballerina/io;
 
 // This function takes one required parameter, one defaultable parameter and
-// one rest parameter of type string. A function can have only one rest parameter.
-// The rest parameter can take any number of values, and is equivalent to an array of the same type. 
+// one rest parameter of type string. A function can have only one rest
+// parameter. The rest parameter can take any number of values, and is
```
[the remark this answers is not in the document]
Comment (NatashaWso2) [reply examples/functions-with-rest-parameter/functions_with_rest_parameter.bal]: @samgnaniah here we need to use 'rest' in lowercase.

File: examples/functions-with-rest-parameter/functions_with_rest_parameter.bal [hunk 19]
```diff
@@ -1,8 +1,9 @@
 import ballerina/io;
 
 // This function takes one required parameter, one defaultable parameter and
-// one rest parameter of type string. A function can have only one rest parameter.
-// The rest parameter can take any number of values, and is equivalent to an array of the same type. 
+// one rest parameter of type string. A function can have only one rest
```
[the remark this answers is not in the document]
Comment (NatashaWso2) [reply examples/functions-with-rest-parameter/functions_with_rest_parameter.bal]: @samgnaniah here we need to use 'rest' in lowercase.

File: examples/integer-ranges/integer_ranges.bal [hunk 20]
```diff
@@ -1,21 +1,23 @@
 import ballerina/io;
 
 public function main() {
-    // A closed integer range `x ... y` generates an array of integers containing values `x` to `y`, `x` and `y`
-    // inclusive
+    // A closed integer range `x ... y` generates an array of integers
+    // containing values `x` to `y`, including `x` and `y`.
     int[] closedRangeArray = 25 ... 28;
+    int closedArrayLength = closedRangeArray.length();
```
Comment (MaryamZi) [inline examples/integer-ranges/integer_ranges.bal]: IMO `closedArrayLength` and `halfOpenArrayLength` sound like closed and half open are properties related to the arrays.

Shall we just say `arrayLength` and use the same variable below too?

File: examples/length/length.bal [hunk 21]
```diff
@@ -71,6 +70,16 @@ public function main() {
...
     io:println("Row count in table : ", length);
 }
+
+type Student record {
+    int id;
+    string fname;
+    string lname;
+    int age;
+    string...
+};
```
Comment (MaryamZi) [inline examples/length/length.bal]:
```suggestion
};

```

File: examples/workers/workers.description [hunk 22]
```diff
@@ -1,5 +1,4 @@
-// Workers in Ballerina allow developers to delegate their tasks to parallel
-// running executions.
-// Workers can be declared within functions or resources.
-// Worker execution starts immediately after invoking the relevant function,
-// remote function, or resource that encloses the worker(s).
+// Workers in Ballerina allow developers to delegate their tasks to parallel running executions.
+// Workers can be declared within functions, remote functions or resources.
```
Comment (samgnaniah) [inline examples/workers/workers.description]:
```suggestion
// Workers can be declared within functions, remote functions, or resources.
```

File: examples/workers/workers.description [hunk 23]
```diff
@@ -1,5 +1,4 @@
...
+// Worker execution starts immediately after executing the statements declared before the first worker declaration in the relevant function,
+// remote function or resource that encloses the worker(s).
```
Comment (samgnaniah) [inline examples/workers/workers.description]:
```suggestion
// remote function, or resource that encloses the worker(s).
```

File: examples/documentation/documentation.bal [hunk 24]
```diff
@@ -20,16 +20,16 @@ public type DummyObject abstract object {
...
+    #              `doThatOnObject` function.
+    # + return - This is the description for the return value of
+    #            `doThatOnObject` function.
     public function doThatOnObject(string paramOne) returns (boolean);
 };
 
 // This is the documentation attachment for record `DummyType`.
-# `DummyType` is a user defined record type in Ballerina program.
-# This `DummyType` is used to hold a string type data field and an int type
+# `DummyType` is a user defined record type in your Ballerina program.
```
Comment (MaryamZi) [inline examples/documentation/documentation.bal]: Do we need to say `your`? Same for other changes in this PR too.

I don't think we've used `your` in general in BBEs.

File: examples/documentation/documentation.description (hunk 15 above)
Comment (MaryamZi) [inline examples/documentation/documentation.description]:
```suggestion
// Documentation attachments allows you to document different Ballerina constructs such as functions, records, objects, listeners, etc. defined in your modules.
```

File: examples/documentation/documentation.description [hunk 25]
```diff
@@ -1,2 +1,2 @@
-// Documentation attachments allows you to document different Ballerina constructs such as functions, records, objects, endpoints etc.
-// These attachments will be used by Docerina to generate API documentation for your modules.
\ No newline at end of file
+// Documentation attachments allows you to document different Ballerina constructs such as functions, records, objects, endpoints etc. defined in your modules.
+// These attachments are used by Docerina (Documentation Generation Tool) to generate API documentation for your modules.
```
Comment (MaryamZi) [inline examples/documentation/documentation.description]:
```suggestion
// These attachments are used by Docerina (the Ballerina documentation generation tool) to generate API documentation for your modules.

```

File: examples/functions/functions.bal [hunk 26]
```diff
@@ -1,20 +1,24 @@
 import ballerina/io;
 
-// This function takes a `string` argument, but does not have a return value.
+// This function takes in a `string` argument, but does not return a value.
```
Comment (MaryamZi) [inline examples/functions/functions.bal]: do we need to add `in`?

File: examples/length/length.bal [hunk 27]
```diff
@@ -45,21 +37,28 @@ public function main() {
     length = myStr.length();
     io:println("String size: ", length);
 
+    // This creates a XML element.
```
Comment (MaryamZi) [inline examples/length/length.bal]:
```suggestion
    // This creates an XML element.
```

File: examples/length/length.description [hunk 28]
```diff
@@ -1,2 +1,2 @@
-// Ballerina supports the `length()` builtin function that returns the length/size of the given value.
+// Ballerina supports the builtin function `length()`  that returns the length/size of the given value.
```
Comment (MaryamZi) [inline examples/length/length.description]: shouldn't this be method ? IINM, we say built-in method right? 

https://docs.google.com/presentation/d/1ZnykKSai8WFJzT6GMIuj1s-6bMwGYaawm_Q1E4_PBa4/edit#slide=id.g424ccecdb3_0_55

File: examples/string-template/string_template.description [hunk 29]
```diff
@@ -1,3 +1,3 @@
 // String templates are string literals that allow embedded expressions.
 // The placeholders for expressions are indicated by two open and close curly braces `{{}}`.
-// You can enclose a string literal with the keyword `string` and two back-tick characters.
\ No newline at end of file
+// You can enclose a string literal with the keyword `string` followed by two back-tick characters.
```
Comment (MaryamZi) [inline examples/string-template/string_template.description]:
```suggestion
// You can enclose a string literal with the keyword `string` followed by two back-tick characters.

```

File: examples/var/var.bal (hunk 12 above)
Comment (MaryamZi) [inline examples/var/var.bal]: Shall we add spaces after //

File: examples/worker-interaction/worker_interaction.bal [hunk 30]
```diff
@@ -2,50 +2,64 @@ import ballerina/io;
...
-        // Send messages to worker `w2`. This message contains a tuple value with member types of `int` and `float`.
+
+        // Send messages asynchronously to worker `w2`. This message contains a
+        // tuple value with member types of `int` and `float`.
         (i, k) -> w2;
-        // Receive a message from worker `w2`. This message contains a `json` type value.
+        io:println("[w1 -> w2] i: ", i, " k: ", k);
+
+        // Receive a message from worker `w2`. This message contains a `json`
+        // type value.
```
Comment (MaryamZi) [inline examples/worker-interaction/worker_interaction.bal]: should this be typed?

File: examples/worker-interaction/worker_interaction.bal [hunk 31]
```diff
@@ -2,50 +2,64 @@ import ballerina/io;
...
+        // until worker `w2` receives the message.
         () send = i ->> w2;
-        // Synchronous send returns nil for successful send or returns error or panic based on receiving worker's state.
+
+        // Synchronous send returns nil if the message is sent successfully or
+        // returns an error or panics based on the receiving worker's state.
         io:println("[w1 ->> w2] successful!!");
 
-        //Send few messages to `w3`.
+        //Send messages asynchronously to worker `w3`.
```
Comment (MaryamZi) [inline examples/worker-interaction/worker_interaction.bal]:
```suggestion
        // Send messages asynchronously to worker `w3`.
```

File: examples/length/length.bal (hunk 27 above)
Comment (MaryamZi) [reply examples/length/length.bal]: Also wondering if we should remove "This" in all the comments and change them to 
```
// Create an integer array with several integer elements.
// Create an XML element.
```
etc.

Maybe we can do this separately.

File: examples/functions/functions.bal (hunk 5 above)
Comment (MaryamZi) [reply examples/functions/functions.bal]: Shall we say
```ballerina
// A public function named `main` is considered a default entry point to a Ballerina program.
```

**Pull request merged**